src/main/java/dk/kb/present/DSOrigin.java

```diff
@@ -198,6 +198,49 @@ public Stream<DsRecordDto> getDSRecords(
         }
     }
 
+    /**
+     * Returns a stream of records with no information on record types or relations between different records.
+     * All types of records are returned.
+     * <p>
+     * The logic is complicated by the need to check for access to the IDs:
+     * The raw stream of records is split into batches in order to lower the amount of external calls to ds-license.
+     * The {@code flatMap(accessFilter)} processes such a batch (a list of {@code DsRecordDto}s) and flattens the
+     * result to a regular stream of {@code DsRecordDto}s.
+     * @param mTime  point in time (epoch * 1000) for the records to deliver, exclusive.
+     * @param maxRecords the maximum number of records to deliver. -1 means no limit.
+     * @param format the format of the record. See {@link #getViews()} for available formats.
+     * @param accessFilter filters which records should be delivered.
+     * @return a stream of records in the requested format.
+     * @throws ServiceException if anything went wrong during construction of the stream.
+     */
+    public Stream<DsRecordDto> getDSRecordsAll(
+            Long mTime, Long maxRecords, String format, Function<List<DsRecordDto>, Stream<DsRecordDto>> accessFilter) {
+        View view = getView(format);
+        log.debug("Extracting with the following view: '{}'", view);
+        log.debug("Calling storage.getDSRecords(origin='{}', mTime={}, maxRecords={})",
+                origin, mTime, maxRecords);
+        try {
+            // 35 is a magic number, which is poor code style. Currently, it controls batch size against ds-license
+            return ExtractionUtils.splitToLists(
+                            storage.getDSRecords(origin, mTime, maxRecords), 35)
+                    .flatMap(accessFilter)
+                    .peek(record -> {
+                        try {
+                            record.data(view.apply(record));
+                        } catch (Exception e) {
+                            throw new RuntimeTransformerException(
+                                    "Exception transforming record '" + record.getId() + "' to format '" + format + "'");
+
+                        }
+                    });
```

Comment:
We have the same (ugly & made by me) piece of code below. Could you extract a `safeView`-method that takes care of the transformation and the exception handling?